### ISSUE TITLE
Discrepancy between debug build & release build

### ISSUE THREAD

**Issue by smeekmeister** [invertase/react-native-firebase#2046]
There is a discrepancy in Firestore between a debug build and release build concerning the document reference. In a release build the "id" field is not part of the document snapshot.

One has to manually extract it from ._documentPath._parts (see example in code)

const storeId = item.store.reference._documentPath._parts[1]
instead of just being able to put  item.store.reference.id

<Carousel
                layout="stack"
                useScrollView={false}
                layoutCardOffset={7}
                data={orderItems}
                sliderWidth={sliderWidth}
                itemWidth={sliderWidth}
                removeClippedSubviews={false}
                onBeforeSnapToItem={(activeSlideIndex) => setSlideIndex(activeSlideIndex)}
                renderItem={({ item, index }) => {
                    const storeId = item.store.reference._documentPath._parts[1]
                    return (
                        <View style={styles.slide}>
                            <Order
                                key={item.id}
                                orderId={order.id}
                                storeId={storeId}
                                podId={podId}
                                orderItem={item}
                                isVisible={index === slideIndex || index === slideIndex + 1}
                            />
                        </View>
                    )
                }}
            />

**Comment by Salakar** [comment]
Hey @smeekmeister - thanks for the report; please could you provide a full reproducible example that demonstrates this behaviour and I'll get it looked at asap.

**Comment by smeekmeister** [comment]
Hi, I have upgraded to v 5.3.0 and now it seems to work. Could it be that this was not the case in an older version. (v.5.1.0)
I have now tested both scenarios
const storeId = item.store.reference._documentPath._parts[1]
as well as
const storeId = item.store.reference.id
and both scenarios seem to work both in debug build and release build.
Back in December we had to dig in the native code to find out that id was not included.
